fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -1389,6 +1389,11 @@ private void applyPeriodicAccruals(final Collection<LoanTransaction> accruals) {
         }
     }
 
+    private LocalDate getDateForRangeCalculationWhenChargeAccrualOnSubmittedDate(LoanTransaction loanTransaction) {
+        return loanTransaction.getLoanChargesPaid().size() == 0 ? loanTransaction.getTransactionDate()
```

Comment:
`isEmpty` might be better for readability.

However I dont really like this method. There are 2 places you check whether accrual transaction date to be used or the charge due date. Would you mind merge the conditions into just one place? You need a date , the method should decide entirely what date to be retrieved and hence to be used.